### PR TITLE
Added 'lifetime' keyword and missing types

### PR DESCRIPTION
**Pull request by rahulghangas**

### CODE REVIEW

**Comment by rahulghangas** [comment]
Tested using [highlight](https://gitlab.com/saalen/highlight) and the [lulesh file](https://github.com/chapel-lang/chapel/blob/master/test/release/examples/benchmarks/lulesh/lulesh.chpl)

File: highlight/highlight/langDefs/chpl.lang [hunk 1]
```diff
@@ -15,26 +15,26 @@ Keywords={
          "begin", "borrowed", "break", "by",
          "catch", "class", "cobegin", "coforall",
          "config", "const", "continue",
-         "delete", "dmapped", "do", "domain",
+         "delete", "dmapped", "do",
          "else", "enum", "except", "export", "extern",
          "false", "for", "forall",
-         "if", "in", "index", "inline", "inout", "iter",
-         "label", "lambda", "let", "local",
+         "if", "in", "index", "init", "init=", "inline", "inout", "iter",
```
Comment (lydia-duncan) [inline highlight/highlight/langDefs/chpl.lang]: Ditto other PR.  I don't think `init` and `init=` should be highlighted, but I think the other changes are right

Comment (lydia-duncan) [reply highlight/highlight/langDefs/chpl.lang]: And again, sorry for the delay and thanks for the contribution!

Comment (rahulghangas) [reply highlight/highlight/langDefs/chpl.lang]: @lydia-duncan I've updated it

**Pull request merged**